main/lsp/tools/make_lsp_types.cc

```diff
@@ -1279,9 +1282,13 @@ void makeLSPTypes(vector<shared_ptr<JSONClassType>> &enumTypes, vector<shared_pt
                        "LSPFileUpdates updates;",
                    });
 
+    auto SorbetTypecheckRunStatus =
+        makeIntEnum("SorbetTypecheckRunStatus", {{"Started", 0}, {"Cancelled", 1}, {"Ended", 2}}, enumTypes);
+
     auto SorbetTypecheckRunInfo = makeObject("SorbetTypecheckRunInfo",
                                              {
-                                                 makeField("tookFastPath", JSONBool),
+                                                 makeField("status", SorbetTypecheckRunStatus),
+                                                 makeField("fastPath", JSONBool),
```

Comment:
Trivial naming observation: It totally makes sense that `tookFastPath` => `fastPath` reflects that if the Status is merely "Started", you don't know which one you'll take.  Is `filesTypeChecked` actually the files that have already been successfully typechecked, or the full list that will be checked?  (This name makes me think that it's the former.)

Reply:
It's the former. It's not specified at the start of typechecking, because (at least for fast path) we don't know which files you'll pull in yet.